perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -297,6 +300,36 @@ public Member createSpecificMember(PerunSession sess, Vo vo, Candidate candidate
 		return member;
 	}
 
+	public Member createSponsoredAccount(PerunSession sess, Map<String, String> params, String namespace, ExtSource extSource, String extSourcePostfix, User owner, Vo vo) throws InternalErrorException, ExtendMembershipException, AlreadyMemberException, WrongReferenceAttributeValueException, WrongAttributeValueException, GroupOperationsException, PasswordCreationFailedException, ExtSourceNotExistsException, LoginNotExistsException, UserNotExistsException {
+		String loginNamespaceUri = "urn:perun:user:attribute-def:def:login-namespace:" + namespace;
+		boolean passwordPresent = params.get("password") != null;
```

Comment:
This might not be clear but I used `get` to check if map contains the key and corresponding value is not null in one call, `containsKey` doesn't check for null value